src/Rpc/NodeGuardService.cs

```diff
@@ -264,25 +282,41 @@ IFMUTXORepository fmutxoRepository
         }
         catch (NoUTXOsAvailableException)
         {
+            CancelWithdrawalRequest(withdrawalRequest);
             _logger.LogError("No available UTXOs for wallet with id {walletId}", request.WalletId);
-            throw new RpcException(new Status(StatusCode.Internal, "No available UTXOs for wallet"));
+            throw new RpcException(new Status(StatusCode.ResourceExhausted, "No available UTXOs for wallet"));
+        }
+        catch (NotEnoughBalanceInWalletException e)
+        {
+            _logger.LogError(e.Message);
+            throw new RpcException(new Status(StatusCode.ResourceExhausted, e.Message));
+        }
```

Comment:
Doesn't it cancel when there are not enough funds?

Reply:
It's not needed because that error comes from the AddAsync. So there is no withdrawal request to cancel yet

Reply:
Thanks!